RedBeanPHP/OODB.php

```diff
@@ -539,6 +539,14 @@ public function getCurrentRepository()
 		return $this->repository;
 	}
 
+	/**
+	 * Return the key maps cache
+	 */
+	public function &getKeyMapsCache()
```

Comment:
Nah, style-wise maybe better get/set - but this is probably better for performance, otherwise the values have to be copied.

Reply:
Speaking about performance, it might be an idea to have a function to 'set' a pre-generated cache map. I'm not sure dynamic caching is even that useful, since PHP script typically cannot 'share' stuff. This is already possible with your solution, but it would be nice to explain it in the comments so we can have it in the automatically generated API docs.

Reply:
Performance-wise I'm not sure how references fare to be honest.

The bean could call a function that would fill the keyMaps instead of doing it directly in `getOwnList` and `&__get` I suppose. I could also make more sense when reading the code.